### PR TITLE
docs(jsdoc): Bearer Authentication Middleware

### PR DESCRIPTION
**Pull request by goisaki**
This PR is to add JSDoc for Bearer Auth Middleware.
Note that the target of the PR is not `main`.

Related:
- #1338
- #2680

- [x] `bun denoify` to generate files for Deno
- [x] `bun run format:fix && bun run lint:fix` to format the code

### CODE REVIEW

File: deno_dist/middleware/bearer-auth/index.ts [hunk 1]
```diff
@@ -23,6 +23,35 @@ type BearerAuthOptions =
...
+ * @see {@link https://hono.dev/middleware/builtin/bearer-auth}
+ *
+ * @param {BearerAuthOptions} options - The options for the bearer authentication middleware.
+ * @param {string | string[]} [options.token] - The string or array of strings to validate the incoming bearer token against.
+ * @param {Function} [options.verifyToken] - The function to verify the token.
+ * @param {string} [options.realm=""] - The domain name of the realm, as part of the returned WWW-Authenticate challenge header.
+ * @param {string} [options.prefix="Bearer"] - The prefix (or known as `schema`) for the Authorization header value.
+ * @param {string} [options.headerName=Authorization] - The header name.
+ * @param {Function} [options.hashFunction] - A function to handle hashing for safe comparison of authentication tokens.
+ * @returns {MiddlewareHandler} - The middleware handler function.
```
Comment (yusukebe) [inline deno_dist/middleware/bearer-auth/index.ts:38]: The hyphen is not necessary.

CleanShot 2024-05-17 at 06 58 55@2x


So it should be:

`* @returns {MiddlewareHandler} The middleware handler function.`

**Comment by yusukebe** [comment]
@goisaki 

Thanks for the PR. I've left a comment.

File: deno_dist/middleware/bearer-auth/index.ts (hunk 1 above)
Comment (goisaki) [reply deno_dist/middleware/bearer-auth/index.ts]: Resolved and squashed to first commit

**Review by yusukebe — APPROVED** [review]
LGTM!

**Pull request merged**